althea_types/src/interop.rs

```diff
@@ -572,6 +573,8 @@ pub struct HardwareInfo {
     pub entire_system_kernel_version: String,
     /// Vector of eth data i.e. whether a link is up and if so what the link speed is
     pub ethernet_stats: Option<Vec<EthernetStats>>,
+    // Vector of wifi devices on router with staion and survey data for each
+    pub wifi_devices: Option<Vec<WifiDevice>>,
```

Comment:
I believe I used an option here for the ethernet stats vec for backwards compatibility with routers that are not yet updated to send this data. In terms of actual representation an empty vec is just as good as a 'None' and requires less nesting to process.

So just flagging that this is the right call for now, but we should probably remove the option at some point (that point may be now for the ethernet stats, but that's a matter for another pr.

Reply:
fixed